File: onyx-database-tests/src/main/java/pojo/Simple.java
package pojo;

import java.io.Serializable;

public class Simple implements Serializable
{
    public Simple()
    {

    }
    public int hiya = 3;

    public int hashCode()
    {
        return hiya;
    }

    public boolean equals(Object val)
    {
        if(val instanceof Simple)
        {
            return ((Simple) val).hiya == hiya;
        }

        return false;
    }
}

File: onyx-database-tests/src/main/java/pojo/Simple.kt
package pojo

import java.io.Serializable

class Simple : Serializable {
    var hiya = 3

    override fun hashCode(): Int {
        return hiya
    }

    override fun equals(`val`: Any?): Boolean {
        return if (`val` is Simple) {
            `val`.hiya == hiya
        } else false

    }
}
